src/Rings/oscar_singular.jl

```diff
@@ -72,6 +72,11 @@ end
 function preimage(f::OscarSingularCoefficientRingMapGeneric, a::Singular.n_unknown)
   parent(a) !== codomain(f) && error("Element not in codomain")
   b = Singular.libSingular.julia(Singular.libSingular.cast_number_to_void(a.ptr))
+  if b isa Singular.FieldElemWrapper || b isa Singular.RingElemWrapper
+    # handle immutable ring elements (such as QQAbFieldElem) which are
+    # put in a wrapper on the Singular side
+    return b.data::elem_type(domain(f))
+  end
```

Comment:
Can you confirm that this is captured by the corresponding test?

Reply:
I added this because a test was failing which needs this change, if that is what you mean with "captured by the test"?

Reply:
Yes, thanks.